### DISCUSSION TITLE
How do I request that an alternative keyword be attributed to an existing language?

### DISCUSSION THREAD

**Discussion by beedell-roke** [github-linguist/linguist#7146] [Classification] [answered]
I don't know which issue template to use to ask that `jCard` be a synonym of `JSON`.

**Comment by lildude** [comment, marked as answer]
You can do it yourself by following the instructions for [adding a new extension](https://github.com/github-linguist/linguist/blob/main/CONTRIBUTING.md#adding-an-extension-to-a-language), but instead add to `aliases`, and add an extension whilst you're at it if it has a specific extension. As with a new extension, you will need to prove extensive usage across GitHub.

**Reply by beedell-roke** [reply]
@lildude, I'm not asking to add a new syntax, so there's no need for a new extension. I'm asking for the `jCard` syntax highlighting string (used like `~~~jCard`) to correspond to the existing `~~~JSON` one.

**Reply by lildude** [reply]
I understand. That's why I said:

> You can do it yourself by following the instructions for [adding a new extension](https://github.com/github-linguist/linguist/blob/main/CONTRIBUTING.md#adding-an-extension-to-a-language), but instead add to `aliases`

It's the aliases that govern what you can use in codeblocks when not using the actual language name. You'd be adding an alias to the existing JSON entry.

**Reply by beedell-roke** [reply]
> https://github.com/github-linguist/linguist/discussions/7146#discussioncomment-11447797

@lildude, thanks. Unlike a specific language, which shall be utilized across GitHub irrespective of whether it is syntax-highlighted, the same shall not apply to aliases. Would it not be reasonable to expect that an alias would solely be utilized if it is *already* syntax-highlighted?

**Reply by lildude** [reply]
I don't understand your question. Aliases allow syntax highlighting to apply to codeblocks using different names.

The following are all JSON, but using different aliases defined here:

https://github.com/github-linguist/linguist/blob/5fad8d57605a914026a65b0e3ff6815d739944de/lib/linguist/languages.yml#L3270-L3274

- json - the language name:

```json
{
  "foo": "bar"
}
```

- jsonl - the second alias (we can't use geojson as it'll try render):

```jsonl
{
  "foo": "bar"
}
```

- sarif - the third alias

```sarif
{
  "foo": "bar"
}
```

As you can see, they're all highlighted as JSON which is what I understand you are wanting to achieve with jCard.

This of course won't apply to any files. It will only apply to codeblocks.

**Reply by beedell-roke** [reply]
> Aliases allow syntax highlighting to apply to codeblocks using different names.

@lildude, I'm aware of that. Thanks, though.

I'm asking why a proposal to add an alias requires more than the request merely being sensible. Specifically, you state that it requires evidence of previous widespread usage of the alias (presumably, in code blocks). However, nobody shall be using an alias for syntax highlighting across GitHub *until* that alias has been implemented in Linguist, so it's a chicken-and-egg situation.

**Reply by lildude** [reply]
Oh, I understand now. Not quite chicken and egg… many of the aliases added on their own over the years have already got widespread usage by the time they're added as they often form part of issues and discussions also taking place.

I'd expect the same for jcard as I'd imagine that's what has prompted you to ask, else you could use JSON.
